server/api/src/rsGeneralAdmin.cpp

```diff
@@ -52,6 +53,63 @@ namespace
         return std::find_if(b, e, [](unsigned char _ch) { return ::isspace(_ch); }) != e;
     } // contains_whitespace
 
+     auto throw_if_inappropriate_rodsadmin_downgrade(
```

Comment:
After reading through the function, I suspect we can add a check at the beginning that causes it to return immediately if `_new_value` is `"rodsadmin"`.

If `_new_value` is `"rodsadmin"`, then there's nothing to do.

This also means the checks near the end of the function handling `"rodsadmin"` shouldn't be needed anymore because, at that point, we know we're doing a downgrade of the user's type.